x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/field_stats_flyout/field_stats_flyout.tsx

```diff
@@ -0,0 +1,178 @@
+    () =>
+      start && end
+        ? { from: moment(start).toISOString(), to: moment(end).toISOString() }
+        : undefined,
+    [start, end]
+  );
+
+  const fieldForStats = useMemo(
+    () => (isDefined(fieldName) ? jobCreator.dataView.getFieldByName(fieldName) : undefined),
+    [fieldName, jobCreator.dataView]
+  );
+
+  const showFieldStats = timeRange && isDefined(jobCreator.dataViewId) && fieldForStats;
+
+  return showFieldStats ? (
+    <FieldStats
+      services={fieldStatsServices}
+      dslQuery={jobCreator.query ?? defaultDatafeedQuery}
+      fromDate={timeRange.from}
+      toDate={timeRange.to}
+      dataViewOrDataViewId={jobCreator.dataView}
+      field={fieldForStats}
+      data-test-subj={`jobCreatorFieldStatsPopover ${fieldForStats.name}`}
+    />
+  ) : null;
+};
+
+interface MLJobWizardFieldStatsFlyoutProps {
+  isFlyoutVisible: boolean;
+  setIsFlyoutVisible: (v: boolean) => void;
+  toggleFlyoutVisible: () => void;
+  setFieldName: (v: string | undefined) => void;
+  fieldName?: string;
+  setFieldValue: (v: string) => void;
+  fieldValue?: string | number;
+}
+export const MLJobWizardFieldStatsFlyoutContext = createContext<MLJobWizardFieldStatsFlyoutProps>({
+  isFlyoutVisible: false,
+  setIsFlyoutVisible: () => {},
+  toggleFlyoutVisible: () => {},
+  setFieldName: () => {},
+  setFieldValue: () => {},
+});
+
+export const FieldStatsFlyoutTrigger = () => {
+  const { setIsFlyoutVisible, isFlyoutVisible } = useContext(MLJobWizardFieldStatsFlyoutContext);
+
+  return (
+    <EuiToolTip
+      content={i18n.translate(
+        'xpack.ml.newJob.wizard.fieldContextPopover.descriptionTooltipContent',
+        {
+          defaultMessage: 'Show top 10 field values',
+        }
+      )}
+    >
+      <EuiButtonIcon
+        iconType="inspect"
+        onClick={() => {
+          setIsFlyoutVisible(!isFlyoutVisible);
+        }}
+        aria-label={i18n.translate(
+          'xpack.ml.newJob.wizard.fieldContextPopover.topFieldValuesAriaLabel',
+          {
+            defaultMessage: 'Show top 10 field values',
+          }
+        )}
+        data-test-subj={'apmCorrelationsContextPopoverButton'}
+      />
+    </EuiToolTip>
+  );
+};
+export const FieldStatsFlyout = () => {
+  const { setIsFlyoutVisible, isFlyoutVisible, fieldName } = useContext(
+    MLJobWizardFieldStatsFlyoutContext
+  );
+  const pushedFlyoutTitleId = useGeneratedHtmlId({
+    prefix: 'pushedFlyoutTitle',
+  });
+
+  const closeFlyout = useCallback(() => setIsFlyoutVisible(false), []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  if (isFlyoutVisible) {
+    return (
+      <EuiFlyout type="push" size="xs" onClose={closeFlyout} aria-labelledby={pushedFlyoutTitleId}>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="m">
+            <h3 id={pushedFlyoutTitleId}>Field stats</h3>
+          </EuiTitle>
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody
+          css={css`
+            width: 250px;
```

Comment:
Looks like there is not enough space for the distribution chart:

<img width="318" alt="Screenshot 2022-12-14 at 15 36 03" src="https://user-images.githubusercontent.com/1415710/207624221-24877adc-4633-42d1-a996-6d92ff111ed3.png">

Reply:
Updated here  [`43c65b5` (#147322)](https://github.com/elastic/kibana/pull/147322/commits/43c65b5efc206c71af2dcc46afbffd8bcda1b5fa)